parcels/_core/utils/string.py

```diff
@@ -0,0 +1,9 @@
+from keyword import iskeyword, kwlist
+
+
+def _assert_valid_python_varname(name):
+    if name.isidentifier() and not iskeyword(name):
```

Comment:
Could we move the string check into here as well to avoid duplication?


```suggestion
    if not isinstance(name, str):
        raise ValueError(f"Expected `name` to be a string, got {type(name)}.")
    if name.isidentifier() and not iskeyword(name):
```

Reply:
Sure! I'll do that.